packages/@groupby/elements-gb-tracker-plugin/src/gb-tracker-plugin.ts

```diff
@@ -0,0 +1,98 @@
+import { Plugin, PluginMetadata, PluginRegistry } from '@groupby/elements-core';
+import { BEACON_SEARCH, BeaconSearchPayload } from '@groupby/elements-events';
+// eslint-disable-next-line import/no-unresolved
+import { AutoSearchEvent } from 'gb-tracker-client/models';
+// eslint-disable-next-line import/no-unresolved
+import { GbTracker } from 'gb-tracker-client/slim-es';
+
+/**
+ * This plugin is responsible for exposing an [[GbTracker]] instance to Core.
+ */
+export default class GbTrackerPlugin implements Plugin {
+  get metadata(): PluginMetadata {
+    return {
+      name: 'gb-tracker',
+      depends: ['dom_events'],
+    };
+  }
+
+  /**
+   * The value that the [[GbTrackerPlugin]] exposes to the Core entity.
+   */
+  gbTracker: ReturnType<typeof GbTracker>;
+
+  /**
+   * A reference to the registry of plugins for later use.
+   */
+  core: PluginRegistry;
+
+  /**
+   * Name of the events plugin.
+   */
+  eventsPluginName = 'dom_events';
+
+  /**
+   * The [[GbTrackerPlugin]] constructor instantiates an instance of the
+   * [[GbTracker]] client and attaches it to this plugin's `gbTracker`
+   * property.
+   *
+   * @param options The options for instantiating the GbTracker client.
+   */
+  constructor(options: TrackerPluginOptions) {
+    this.triggerSearchBeacon = this.triggerSearchBeacon.bind(this);
+
+    this.gbTracker = new GbTracker(options.customerId, options.area, options.overridePixelUrl);
+  }
+
+  /**
+   * Returns this plugin's instance of the [[GbTracker]] client.
+   */
+  register(plugins: PluginRegistry): ReturnType<typeof GbTracker> {
+    this.core = plugins;
+    return this.gbTracker;
+  }
+
+  /**
+   * Sets event listeners for beacon events. Also calls the GbTracker method
+   * `autoSetVisitor()` to allow for future beacon events to be registered.
+   */
+  ready(): void {
+    this.core[this.eventsPluginName].registerListener(BEACON_SEARCH, this.triggerSearchBeacon);
+
+    this.gbTracker.autoSetVisitor();
```

Comment:
I think you're right - this can go in the constructor. I wanted to be sure about timing, but if we've successfully instantiated a tracker, we can obviously call it as well.